src/index.ts

```diff
@@ -150,23 +174,30 @@ export class Client {
     });
 
     if (resp.status !== 200) {
+      let code;
       let message = resp.statusText;
       try {
-        let response = await resp.json();
-        message = response.message;
+        ({ code, message } = (await resp.json()) as APIError);
       } catch (_) {}
 
+      console.error(
+        `Search request failed due to an error. HTTP code: ${resp.status}, Code: ${code}, Message: ${message}`
+      );
+
       if (resp.status === 403) {
+        console.error(
+          `Check the domain ${window.location.origin} is an authorized query domain. See https://app.search.io/collection/domains`
```

Comment:
`window.location.hostname` gives the info that they need to enter on collection/domains (protocol and port don't matter)

Reply:
@wwalser will adjust, thanks.